diff -urdN pidgin-gnome-keyring-2.0.org/Makefile pidgin-gnome-keyring-2.0/Makefile
--- pidgin-gnome-keyring-2.0.org/Makefile	2015-02-07 04:23:05.000000000 +0100
+++ pidgin-gnome-keyring-2.0/Makefile	2015-09-27 22:17:58.859841783 +0200
@@ -15,7 +15,7 @@
 
 ${TARGET}.so: ${TARGET}.c
 
-	${CC} ${CFLAGS} ${LDFLAGS} -Wall -I. -g -O2 ${TARGET}.c -o ${TARGET}.so -shared -fPIC -DPIC -ggdb ${PURPLEFLAGS} ${SECRETFLAGS} -DVERSION=\"${VERSION}\"
+	${CC} ${CFLAGS} ${LDFLAGS} -I. ${TARGET}.c -o ${TARGET}.so -shared -fPIC -DPIC ${PURPLEFLAGS} ${SECRETFLAGS} -DVERSION=\"${VERSION}\"
 
 install: ${TARGET}.so
 	mkdir -p ${DESTDIR}/usr/lib/purple-2/
